src/library/scala/collection/mutable/Builder.scala

```diff
@@ -30,31 +30,45 @@ trait Builder[-A, +To] extends Growable[A] { self =>
   /** Result collection consisting of all elements appended so far. */
   def result(): To
 
-  /** Gives a hint how many elements are expected to be added
-    *  when the next `result` is called. Some builder classes
-    *  will optimize their representation based on the hint. However,
-    *  builder implementations are still required to work correctly even if the hint is
-    *  wrong, i.e. a different number of elements is added.
-    *
-    *  @param size  the hint how many elements will be added.
-    */
+  /** Gives a hint how many elements are expected to be added in total
+   *  by the time `result` is called.
+   *
+   *  Some builder classes will optimize their representation based on the hint.
+   *  However, builder implementations are required to work correctly even if the hint is
+   *  wrong, i.e., if a different number of elements is added.
+   *
+   *  The semantics of supplying a hint out of range, such as a size that is negative
+   *  or unreasonably large, is not specified but is implementation-dependent.
```

Comment:
I think it's like using a collection for its specific characteristics: a specific builder might take -1 to signal use a default size, etc, or whatever.

The use case was `sizeHint(coll.knownSize)`, or -1. The semantics there is clearly "ignore me, I don't even know my size". So they should have used `sizeHint(coll)` which does ignore it.

Reply:
PS) so if I try to size hint a huge value, should it ignore my spurious input or take the biggest value it can allocate?

It seems to me that is implementation-dependent.

Reply:
I've understood -1 to mean: "no meaningful size information available".

"should it ... take the biggest value it can allocate?"  this looks a lot like a can of worms labelled; "Should we measuring available heap space before allocating more space for a growing array?"

Reply:
But the previous fix was just to "round down" to a reasonable maximum.

I'm not sure why I'm reluctant to "ignore sizeHint < 0". Perhaps because the builder is free to ignore it, there is no reason to specify it, as it is not encoded here.